kpi/tests/api/v1/test_api_users.py

```diff
@@ -1,7 +1,46 @@
 # coding: utf-8
+from django.contrib.auth.models import User
+from rest_framework.reverse import reverse
+
 from kpi.tests.api.v2 import test_api_users
 
 
 class UserListTests(test_api_users.UserListTests):
 
     URL_NAMESPACE = None
+
+    def test_current_user_extra_details_kludges(self):
+        self.client.login(username='someuser', password='someuser')
+        user = User.objects.get(username='someuser')
+        xtradata = user.extra_details.data
+        assert xtradata == {}
+
+        # `primarySector` should be renamed to `sector`
+        xtradata['primarySector'] = 'camelCase Administration'
+        user.extra_details.save()
+        response = self.client.get(reverse('currentuser-detail'))
+        # lone string should be transformed to object with label and value
+        assert response.data['extra_details']['sector'] == {
+            'label': 'camelCase Administration',
+            'value': 'camelCase Administration',
+        }
+        assert not 'primarySector' in response.data['extra_details']
```

Comment:
Shouldn't be `assert 'primarySector' not in response.data['extra_details']` [instead](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)?
:fly: 😗

Reply:
right